sdk/core/Azure.Core/src/OperationOfT.cs

```diff
@@ -0,0 +1,120 @@
+// Licensed under the MIT License.
+
+using System;
+using System.ComponentModel;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Azure
+{
+    /// <summary>
+    /// Represents a long running operation (LRO).
+    /// </summary>
+    /// <typeparam name="T">The final result of the LRO.</typeparam>
+    public abstract class Operation<T>
+    {
+        T _value;
+        Response _response;
+
+        /// <summary>
+        /// Final result of the LRO.
+        /// </summary>
+        /// <remarks>
+        /// This property can be accessed only after the operation completes succesfully (HasValue is true).
+        /// </remarks>
+        public T Value
+        {
+            get {
+                if (!HasValue) throw new InvalidOperationException("operation has not completed");
+                return _value;
+            }
+            protected set
+            {
+                _value = value;
+            }
+        }
+
+        /// <summary>
+        /// The last HTTP response received from the server.
+        /// </summary>
+        /// <remarks>
+        /// The last response returned from the server during the lifecycle of this instance.
+        /// An instance of Operation<typeparamref name="T"/> sends requests to a server in UpdateStatusAsync, UpdateStatus, and other methods.
+        /// Responses from these requests can be accessed using GetRawResponse.
+        /// </remarks>
+        public Response GetRawResponse() => _response;
+        protected void SetRawResponse(Response response) => _response = response;
+
+        /// <summary>
+        /// Returns true if the LRO completed.
+        /// </summary>
+        public abstract bool HasCompleted { get; }
+
+        /// <summary>
+        /// Returns true if the LRO completed succesfully and has produced final result (accessible by Value property).
+        /// </summary>
+        public abstract bool HasValue { get; }
+
+        /// <summary>
+        /// Polling interval used by WaitCompletionAsync method. 
+        /// </summary>
+        /// <remarks>
+        /// The interval can change based on information returned from the server.
+        /// For example, the server might communicate to the client that there is not reason to poll for status change sooner than some time.
+        /// </remarks>
+        public TimeSpan PollingInterval { get; set; } = TimeSpan.FromSeconds(1);
+
+        /// <summary>
+        /// Reriodically calls the server till the LRO completes.
+        /// </summary>
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
+        /// <remarks>
+        /// This method will periodically call UpdateStatusAsync till HasCompleted is true, then return the final result of the operation.
+        /// </remarks>
+        public virtual async ValueTask<Response<T>> WaitCompletionAsync(CancellationToken cancellationToken = default)
+        {
+            while (true)
```

Comment:
Should we remove the default value of the `cancellationToken`?  By forcing users to pass this themselves, they might realize it's the only way to stop this thing from looping for a full week.

Reply:
This would violate our guidelines. Besides, we have many methods in .NET that can block for a very long time and they all have optional cancellation token parameters. If you feel strongly bout it, please bring it as an agenda items to a board meeting.